libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Queue/CSFNetwork.m

```diff
@@ -304,11 +302,11 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
 - (void)receivedDevicedUnauthorizedError:(CSFAction *)action {
 }
 
-#pragma mark - Handling Login State
+#pragma mark - SFAuthenticationManagerDelegate
 
-- (void)userWillLogOutNotification:(NSNotification *)notification {
-    SFUserAccount *userAccount = notification.userInfo[kSFUserAccountKey];
-    [[self class] cleanupNetworkForUserAccount:userAccount];
+- (void)authManager:(SFAuthenticationManager *)manager willLogoutUser:(SFUserAccount *)user {
+    [[self class] removeSharedInstance:user];
```

Comment:
My only concern with this approach is that it gets into potentially tenuous territory.  In this implementation, it's probably fine.  If you had something like this, on the other hand:

```obj-c
- (void)authManager:(SFAuthenticationManager *)manager willLogoutUser:(SFUserAccount *)user {
    [[self class] removeSharedInstance:user];
    self.someOtherProperty = nil;
```

The app would crash, because after the `removeSharedInstance` call, the instance no longer exists, and yet the execution path is still in the instance's code base.

Ideally, it would be a class-level responsibility for consuming the notification and cleaning up.  In practice, this is difficult, because the class itself cannot adopt the delegate protocol.  You could implement a helper class to facilitate that process, but it gets slightly convoluted.

I think we just need to be aware of this fact, and tread lightly, at any rate.